python/chronos/script/benchmark-chronos.py

```diff
@@ -15,153 +15,253 @@
+        print("QUANTIZATION DONE")
 
+    # predict
+    if 'torch' in args.inference_framework:
+        import torch
+        if args.cores: torch.set_num_threads(args.cores)
+        with torch.no_grad():
+            if args.model == 'autoformer':
+                for x, y, x_, y_ in test_loader:
+                    st = time.time()
+                    yhat = forecaster.predict((x.numpy(), y.numpy(), x_.numpy(), y_.numpy()))
+                    latency.append(time.time()-st)
+            else:
+                for x, y in test_loader:
+                    st = time.time()
+                    yhat = forecaster.predict(x.numpy(), quantize=args.quantize)
+                    latency.append(time.time()-st)
+        records['torch_latency'] = stats.trim_mean(latency, latency_trim_portion)
+        records['torch_percentile_latency'] = np.percentile(latency, latency_percentile)
+
+    # predict with onnx
+    if 'onnx' in args.inference_framework:
+        if args.cores: forecaster.build_onnx(thread_num=args.cores)
+        for x, y in test_loader:
+            st = time.time()
+            yhat = forecaster.predict_with_onnx(x.numpy(), quantize=args.quantize)
+            latency_onnx.append(time.time()-st)
+        records['onnx_latency'] = stats.trim_mean(latency_onnx, latency_trim_portion)
+        records['onnx_percentile_latency'] = np.percentile(latency_onnx, latency_percentile)
+
+    # predict with openvino
+    if 'openvino' in args.inference_framework:
+        if args.cores: forecaster.build_openvino(thread_num=args.cores)
+        for x, y in test_loader:
+            st = time.time()
+            yhat = forecaster.predict_with_openvino(x.numpy(), quantize=args.quantize)
+            latency_vino.append(time.time()-st)
+        records['openvino_latency'] = stats.trim_mean(latency_vino, latency_trim_portion)
+        records['openvino_percentile_latency'] = np.percentile(latency_vino, latency_percentile)
+
+def result():
+    print(">>>>>>>>>>>>> test-run information >>>>>>>>>>>>>")
+    print("Model:", args.model)
+    print("Stage:", args.stage)
+    print("Dataset:", args.dataset)
+    if args.cores: print("Cores:", args.cores)
+    else: print("Cores:", psutil.cpu_count(logical=False) * int(subprocess.getoutput('cat /proc/cpuinfo | grep "physical id" | sort -u | wc -l')))
+    print("Lookback:", args.lookback)
+    print("Horizon:", args.horizon)
+
+    if args.stage == 'train':
+        print("\n>>>>>>>>>>>>> train result >>>>>>>>>>>>>")
+        print("avg throughput: {}".format(records['train_throughput']))
+        print(">>>>>>>>>>>>> train result >>>>>>>>>>>>>")
+    elif args.stage == 'latency':
+        for framework in args.inference_framework:
+            print("\n>>>>>>>>>>>>> {} latency result >>>>>>>>>>>>>".format(framework))
+            print("avg latency: {}ms".format(records[framework+'_latency'] * 1000))
+            print("90p latency: {}ms".format(records[framework+'_percentile_latency'][1] * 1000))
```

Comment:
we may add a p50 here. btw, it's p50, p90, p95 and p99. :)

Reply:
has revised